From 00e593acd5f5a450f7d49a088444457e706d9c8a Mon Sep 17 00:00:00 2001
From: Liu Ying <b17645@freescale.com>
Date: Fri, 8 Jan 2010 15:21:22 -0500
Subject: [PATCH 090/280] ENGR00119897 IPU lib:Fix wrong uv-offset for NV12
 pixel format

NV12 is a partial interleaved pixel format, the uv-offset is
calculated wrongly when vertical cropping is needed.

Signed-off-by: Liu Ying <b17645@freescale.com>
---
 ipu/mxc_ipu_hl_lib.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ipu/mxc_ipu_hl_lib.c b/ipu/mxc_ipu_hl_lib.c
index b428990..c8fd370 100644
--- a/ipu/mxc_ipu_hl_lib.c
+++ b/ipu/mxc_ipu_hl_lib.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2009 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2009-2010 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  */
 
@@ -379,7 +379,7 @@ static void _ipu_update_offset(unsigned int fmt, unsigned int width, unsigned in
 		case IPU_PIX_FMT_NV12:
 			*off = pos_y * width + pos_x;
 			*uoff = (width * (height - pos_y) - pos_x)
-				+ width * pos_y + pos_x;
+				+ width * pos_y/2 + pos_x;
 			break;
 		default:
 			*off = (pos_y * width + pos_x) * fmt_to_bpp(fmt)/8;
-- 
1.8.0

